packages/transaction-controller/src/utils/batch.ts

```diff
@@ -364,7 +376,23 @@ async function addTransactionBatchWithHook(
     throw new Error('No publish batch hook provided');
   }
 
+  const chainId = getChainId(networkClientId);
   const batchId = generateBatchId();
+
+  if (requireApproval) {
```

Comment:
Should we also check `useHook` is `true` at this stage as EIP-7702 uses the standard confirmation?

Reply:
Done.